docs/user_guide/reductions.rst

```diff
@@ -0,0 +1,429 @@
+    :math:`\mathbf{\lambda}` and gives it back to the
+    :math:`\mathbf{\lambda}`-player
+#.  The :math:`\mathbf{\lambda}`-player examines how :math:`Q` violates the constraints
+    and proposes a new :math:`\mathbf{\lambda}`
+#.  Continue until the constraints are all satisfied
+
+In the sections below, we will discuss the optimal strategies for each of the
+two players.
+
+
+Strategy for the :math:`\mathbf{\lambda}`-player
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+For a given :math:`Q`, the :math:`\mathbf{\lambda}`-player seeks to maximise
+:math:`L(Q, \mathbf{\lambda})` by picking a valid :math:`\mathbf{\lambda}`
+vector.
+Recalling that this vector is constrained to have positive components and
+:math:`||\mathbf{\lambda}||_1 \le B`, there are two options.
+If no constraints are violated, then the best option is to set all compoments
+of :math:`\mathbf{\lambda} = \mathbf{0}`.
+Otherwise, the constraint on the :math:`\mathcal{l}_1`-norm means that the
+best options is to put all of the weight (i.e. :math:`B`) into the element
+of :math:`\mathbf{\lambda}` corresponding to the most violated constraint.
+
+Formally, if we define:
+
+.. math::
+    \hat{\gamma}(Q) = M \hat{\mathbf{\mu}}(Q)
+
+then the best response of the :math:`\mathbf{\lambda}`-player is given by
+
+.. math::
+    \begin{eqnarray}
+    \mathbf{0} & \qquad & \mbox{if $\hat{\gamma}(Q) \le \hat{\mathbf{c}}$, }\\
+    B \mathbf{e}_{k^*} & \qquad & \mbox{otherwise}
+    \end{eqnarray}
+
+where :math:`\mathbf{e}_k` is the :math:`k^{\mbox{th}}` basis vector for :math:`\mathbf{\lambda}`,
+and :math:`k^* = \operatorname{argmax}_k \left [\hat{\gamma}(Q)_k - \hat{c}_k \right ]`
+is the index of the most-violated constraint.
+
+Strategy for the :math:`Q`\-player
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+For a given :math:`\mathbf{\lambda}`, the :math:`Q`\-player seeks to minimise
+:math:`L(Q, \mathbf{\lambda})` by picking a particular form for the ensemble of
+classifiers :math:`Q`.
+Since :math:`L` is linear in :math:`Q`, this can always be achieved by selecting
+a single classifier :math:`h \in Q`.
+This then gives:
+
+.. math::
+    \begin{eqnarray}
+    L(h, \mathbf{\lambda})
+    & = &
+    \hat{\mbox{err}}(h) + \mathbf{\lambda}^{\mbox{T}}( M \hat{\mathbf{\mu}}(h) - \mathbf{\hat{c}}) \\
+    & = &
+    \hat{\E} \left [ \mathbf{1} {h(X) \ne Y} \right ]
+        - \mathbf{\lambda}^{\mbox{T}}\mathbf{\hat{c}}
+        + \sum_{k,j} M_{k,j} \lambda_k \hat{\mu}_j (h)
+    \end{eqnarray}
+    :label: eq_q_player_lagrange
+
+From equation :eq:`eq_moment_definition` we have:
+
+.. math::
+    \hat{\mu}_j(h) = \hat{\E}
+        \left[ 
+            g_j(X, A, Y, h(X)) | \mathcal{E}_j
+        \right ]
+
+Using the empirical event probabilities
+
+.. math::
+    p_j = \hat{\P} [ \mathcal{E}_j ]
+
+we see that
+
+.. math::
+    \hat{\mu}_j(h) = \frac{1}{p_j}
+        \hat{\E}
+            \left [
+                g_j(X, A, Y, h(X)) \mathbf{1} \{ (X,A,Y) \in \mathcal{E}_j \}
```

Comment:
I feel that this transformation could use an extra bit of explanation. I don't quite feel it's obvious, and I'm not sure how to explain it myself (which indicates that I don't quite understand it myself).